extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayRecorder.java

```diff
@@ -34,23 +34,31 @@ public void doStartActions(FlywayRuntimeConfig config, BeanContainer container)
         if (config.defaultDataSource.migrateAtStart) {
             migrate(container, Default.Literal.INSTANCE);
         }
+        if (config.defaultDataSource.validateOnMigrate) {
+            validate(container, Default.Literal.INSTANCE);
+        }
         for (Entry<String, FlywayDataSourceRuntimeConfig> configPerDataSource : config.namedDataSources.entrySet()) {
             if (configPerDataSource.getValue().cleanAtStart) {
                 clean(container, FlywayDataSource.FlywayDataSourceLiteral.of(configPerDataSource.getKey()));
             }
             if (configPerDataSource.getValue().migrateAtStart) {
                 migrate(container, FlywayDataSource.FlywayDataSourceLiteral.of(configPerDataSource.getKey()));
             }
+            if (configPerDataSource.getValue().validateOnMigrate) {
+                validate(container, FlywayDataSource.FlywayDataSourceLiteral.of(configPerDataSource.getKey()));
```

Comment:
There is no need to call validate again since you are already setting` configure.validateOnMigrate` in `FlywayCreator`

Reply:
OK, I can't probably relate that other than this setting might be a global one rather than per datasource, because I see the other calls being made for `cleat at start` and `migrate`...but I will leave this up to you since you know that better than myself.

In the end, you are stating that: this `if` statement (inside the `for` loop) is unnecessary and can be deleted, right?

Reply:
I believe so, yes. Looking at the Flyway sources it seems thaf validate() is called internally when the `validateOnMigrate` is true